### ISSUE TITLE
MX codegen with division causing compile error

### ISSUE THREAD

**Issue by BachiLi** [casadi/casadi#1554]
The following code will generate "/*" in the c code causing compile error:
```
from casadi import *
x = MX.sym('x', 2)
y = MX.sym('y', 2)
z = x[0] / y

test_division = MXFunction("test_division", [x,y],[z])
test_division.generate()
```
output c code:
```
int test_division(const real_t** arg, real_t** res, int* iw, real_t* w) {
  int i, j, k, *ii, *jj, *kk;
  const int *cii;
  real_t r, s, t, *rr, *ss, *tt;
  const real_t *cr, *cs, *ct;
  const real_t** arg1=arg+2;
  real_t** res1=res+1;
  real_t *w0=w+0, w1;
  if (arg[0])
    copy_n(arg[0], 2, w0);
  else 
    fill_n(w0, 2, 0);
  for (rr=(&w1), ss=w0+0; ss!=w0+1; ss+=1) *rr++ = *ss;
  if (arg[1])
    copy_n(arg[1], 2, w0);
  else 
    fill_n(w0, 2, 0);
  for (i=0, rr=w0, cs=w0; i<2; ++i) *rr++  = (w1/*cs++);
  if (res[0]) copy_n(w0, 2, res[0]);
  return 0;
}
```

Notice the 4th line from the end:

```
*rr++  = (w1/*cs++);
```

Compiler will treat the "/*" as the beginning of a comment section.

**Comment by jgillis** [comment]
Thanks for reporting!

**Comment by BachiLi** [comment]
Thanks for fixing the bug.  Unfortunately, this is not completely fixed when there are constant in the numerator:
```
from casadi import *
y = MX.sym('y', 2)
z = 1.0 / y

test_division = MXFunction("test_division", [y],[z])
test_division.generate()
```
The output c code is:
```
int test_division(const real_t** arg, real_t** res, int* iw, real_t* w) {
  int i, j, k, *ii, *jj, *kk;
  const int *cii;
  real_t r, s, t, *rr, *ss, *tt;
  const real_t *cr, *cs, *ct;
  const real_t** arg1=arg+1;
  real_t** res1=res+1;
  real_t *w0=w+0;
  if (arg[0])
    copy_n(arg[0], 2, w0);
  else 
    fill_n(w0, 2, 0);
  for (i=0, rr=w0, cs=w0; i<2; ++i) *rr++ = (1./*cs++);
  if (res[0]) copy_n(w0, 2, res[0]);
  return 0;
}
```
Notice again the 3rd line from the end:
```
  for (i=0, rr=w0, cs=w0; i<2; ++i) *rr++ = (1./*cs++);
```

**Comment by jgillis** [comment]
See #1581

**Comment by jaeandersson** [comment]
Fixed in 85b92837697eda1910e3795b63897bcc0adb83d0.
